docs/3.0.0-beta.x/migration-guide/migration-guide-alpha.26-to-beta.md

```diff
@@ -0,0 +1,736 @@
+
+#### Config
+
+When customizing configurations you only need to move the modified files.
+
+Strapi merges the plugin's configurations with the ones in `./extensions/pluginName/config`. Hence you should also only add the fields that you modified in your custom configurations files.
+
+**Before**
+
+`./plugins/graphql/config/settings.json`
+
+```json
+{
+  "endpoint": "/graphql", // default
+  "tracing": false, // default
+  "shadowCRUD": true, // default
+  "playgroundAlways": false, // default
+  "depthLimit": 2,
+  "amountLimit": 25
+}
+```
+
+**After**
+
+`./extensions/graphql/config/settings.json`
+
+```json
+{
+  "depthLimit": 2,
+  "amountLimit": 25
+}
+```
+
+#### Routes
+
+If you modified `./plugins/pluginName/config/routes.json` you will have to copy the file to `./extentions/pluginName/config/routes.json` and only keep the routes you customized or added.
+
+All the unchanged routes must be removed from this file.
+
+**Before**
+
+`./plugins/users-permissions/config/routes.json`
+
+```json
+{
+  "routes": [
+    {
+      "method": "GET",
+      "path": "/",
+      "handler": "UsersPermissions.customIndex",
+      "config": {
+        "policies": []
+      }
+    },
+    {
+      "method": "GET",
+      "path": "/init",
+      "handler": "UsersPermissions.init",
+      "config": {
+        "policies": [],
+        "description": "Check if the first admin user has already been registered",
+        "tag": {
+          "plugin": "users-permissions",
+          "name": "Role"
+        }
+      }
+    }
+    //...
+  ]
+}
+```
+
+**After**
+
+`./extensions/users-permissions/config/routes.json`
+
+```json
+{
+  "routes": [
+    {
+      "method": "GET",
+      "path": "/",
+      "handler": "UsersPermissions.customIndex",
+      "config": {
+        "policies": []
+      }
+    }
+  ]
+}
+```
+
+#### Controllers & Services
+
+To migrate controllers and services you must move your customized controllers and services to the `./extensions/pluginName/**/*`, delete all the non customized ones and keep only the methods you modified in your customized files·
+
+For example if you have a `customIndex` action in the `User` controller you only have to create the `./extensions/users-permissions/controllers/User.js` file and keep your `customIndex` action in it. You can delete the rest of the files and methods.
+
+**Before**
+
+`./plugins/users-permissions/controllers/User.js`
+
+```javascript
+module.exports = {
+  customIndex: async ctx => {},
+
+  find: async (ctx, next, { populate } = {}) => {
+    //...
+  },
+
+  me: async ctx => {
+    //...
+  },
+
+  //...
+};
+```
+
+**After**
+
+`./extensions/users-permissions/controllers/User.js`
+
+```js
+module.exports = {
+  customIndex: async ctx => {},
+};
+```
+
+> The same goes for services
+
+#### Models
+
+If you have modified (or cre ated relations) with a plugin's model you will have to move your Model file to `./extensions/pluginName/models/Model.settings.json`
```

Comment:
If you have modified (or **created** relations) with a plugin's model**,** you will have to move your Model file to `./extensions/pluginName/models/Model.settings.json`

Two edits.  including an extra ","